src/SurveyService.Business/Commands/Option/CreateOptionCommand.cs

```diff
@@ -0,0 +1,61 @@
+﻿using FluentValidation.Results;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.SurveyService.Business.Commands.Option.Interfaces;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Option;
+using LT.DigitalOffice.SurveyService.Validation.Option.Interfaces;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.SurveyService.Business.Commands.Option;
+
+public class CreateOptionCommand : ICreateOptionCommand
+{
+  private readonly IHttpContextAccessor _httpContextAccessor;
+  private readonly ICreateOptionRequestValidator _validator;
+  private readonly IOptionRepository _optionRepository;
+  private readonly IDbOptionMapper _mapper;
+  private readonly IResponseCreator _responseCreator;
+
+  public CreateOptionCommand(
+    IHttpContextAccessor httpContextAccessor,
+    ICreateOptionRequestValidator validator,
+    IOptionRepository optionRepository,
+    IDbOptionMapper mapper,
+    IResponseCreator responseCreator)
+  {
+    _httpContextAccessor = httpContextAccessor;
+    _validator = validator;
+    _optionRepository = optionRepository;
+    _mapper = mapper;
+    _responseCreator = responseCreator;
+  }
+
+  public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateOptionRequest request)
+  {
+    ValidationResult validationResult = await _validator.ValidateAsync((request, null));
```

Comment:
In validator you wait for ValueType<CreateOptionRequest, bool?>, where bool stands for "hasCustomOptions", but here you just always pass null??

Reply:
"Option" can be created separately (this PR) or with "Question" (one question has many options). If Option is created separately, into validator goes "null": it shows, which case of validation should be picked and shows that value "hasCustomOptions"(it is question's property) should be received from DB.
If Option is created with Question, into validator goes "hasCustomOptions" from request.
Idea is to make an universally validator for Option.

Reply:
But we accept in validator hasCustomOption as the second paramert of incomming struct, but we always pass null from command and never change it anywhere else?

Reply:
null in the parameters allows to distinguish between the request from the command and request from the question validator